common/src/main/java/tachyon/underfs/UnderFileSystem.java

```diff
@@ -364,6 +368,53 @@ public abstract OutputStream create(String path, short replication, int blockSiz
   public abstract String[] list(String path) throws IOException;
 
   /**
+   * Returns an array of strings naming the files and directories in the directory denoted by this
+   * abstract pathname, and all of its subdirectories.
+   *
+   * <p>
+   * If this abstract pathname does not denote a directory, then this method returns {@code null}.
+   * Otherwise an array of strings is returned, one for each file or directory in the directory and
+   * its subdirectories. Names denoting the directory itself and the directory's parent directory
+   * are not included in the result. Each string is a path relative to the given directory.
+   *
+   * <p>
+   * There is no guarantee that the name strings in the resulting array will appear in any specific
+   * order; they are not, in particular, guaranteed to appear in alphabetical order.
+   *
+   * @param path the path to list
+   * @return An array of strings naming the files and directories in the directory denoted by this
+   *         abstract pathname and its subdirectories. The array will be empty if the directory is
+   *         empty. Returns {@code null} if this abstract pathname does not denote a directory, or
+   *         if an I/O error occurs.
```

Comment:
It probably makes sense to take out the 'if an I/O error occurs' part in both methods. So we should just return null if the path denotes a file.